tasklist/client/src/v1/TaskDetailsLayout/AssignButton/index.tsx

```diff
@@ -29,156 +33,109 @@ const getAssignmentToggleLabels = () =>
-  const [assignmentStatus, setAssignmentStatus] = useState<AssignmentStatus>(
-    () => (taskState === 'ASSIGNING' ? 'assigning' : 'off'),
-  );
-
+const AssignButton: React.FC<Props> = ({id, assignee, taskState}) => {
+  const isAssigned = typeof assignee === 'string' && taskState !== 'ASSIGNING';
   const {t} = useTranslation();
+  const [assignmentStatus, setAssignmentStatus] =
+    useState<AssignmentStatus>('off');
+  const {mutateAsync: pollForAssignmentResult} = usePollForAssignmentResult();
   const {mutateAsync: assignTask, isPending: assignIsPending} = useAssignTask();
   const {mutateAsync: unassignTask, isPending: unassignIsPending} =
     useUnassignTask();
   const isLoading =
     (assignIsPending || unassignIsPending || taskState === 'ASSIGNING') ??
     false;
 
+  const effectiveAssignmentStatus =
+    taskState === 'ASSIGNING' ? 'assigning' : assignmentStatus;
+
+  function getAsyncActionButtonStatus() {
+    if (isLoading || effectiveAssignmentStatus !== 'off') {
+      const ACTIVE_STATES: AssignmentStatus[] = ['assigning', 'unassigning'];
+
+      return ACTIVE_STATES.includes(effectiveAssignmentStatus) ||
+        taskState === 'ASSIGNING'
+        ? 'active'
+        : 'finished';
+    }
+
+    return 'inactive';
+  }
+
   const handleAssignmentClick = async () => {
     try {
       setAssignmentStatus('assigning');
       await assignTask(id);
+
+      await pollForAssignmentResult({
+        taskId: id,
+        wasAssigned: false,
+      });
+
       setAssignmentStatus('assignmentSuccessful');
       tracking.track({eventName: 'task-assigned'});
     } catch (error) {
-      if (!(error instanceof Error)) {
+      const {data: parsedError, success} = requestErrorSchema.safeParse(error);
+
+      if (success && parsedError.variant === 'failed-response') {
+        const errorData = await parsedError.response.json();
+
         notificationsStore.displayNotification({
           kind: 'error',
           title: t('taskDetailsTaskAssignmentError'),
           isDismissable: true,
         });
 
-        setAssignmentStatus('off');
-        return;
-      }
-
-      if (error.name === assignmentErrorMap.taskProcessingTimeout) {
-        tracking.track({
-          eventName: 'task-assignment-delayed-notification',
-        });
+        if (errorData.title !== 'DEADLINE_EXCEEDED') {
+          setAssignmentStatus('off');
+        }
         return;
       }
 
+      notificationsStore.displayNotification({
+        kind: 'error',
+        title: t('taskDetailsTaskAssignmentError'),
+        isDismissable: true,
+      });
       setAssignmentStatus('off');
-      if (error.name === assignmentErrorMap.invalidState) {
-        tracking.track({
-          eventName: 'task-assignment-rejected-notification',
-        });
-        notificationsStore.displayNotification({
-          kind: 'error',
-          title: t('taskDetailsTaskAssignmentError'),
-          subtitle: t('taskDetailsTaskAssignmentRejectionErrorSubtitle'),
```

Comment:
`taskDetailsTaskAssignmentRejectionErrorSubtitle` needs to be removed from the translation files if it is not needed anymore